src/PolygonDraw/Map.tsx

```diff
@@ -316,31 +331,55 @@ export class BaseMap extends React.Component<Props, State> {
         );
     };
 
-    renderPolygonPoints = () => this.props.polygonCoordinates.map(this.renderPolygonVertex);
+    renderActivePolygonPoints = () => {
+        return this.props.polygonCoordinates[this.props.activePolygonIndex].map(this.renderPolygonVertex);
+    };
 
     renderVertexEdge = (coordinate: Coordinate, index: number) => (
-        <EdgeVertex
-            key={index}
-            index={index}
-            coordinate={coordinate}
-            onClick={this.dispatch.addPointToEdge}
-        />
+        <EdgeVertex key={index} index={index} coordinate={coordinate} onClick={this.props.addPointToEdge} />
     );
 
-    renderPolygonEdges = () => getPolygonEdges(this.props.polygonCoordinates).map(this.renderVertexEdge);
+    renderPolygonEdges = () => {
+        return getPolygonEdges(this.props.polygonCoordinates[this.props.activePolygonIndex]).map(this.renderVertexEdge);
+    };
 
-    renderPolygon = () => (
-        <Polygon
-            positions={this.props.polygonCoordinates.map(createLeafletLatLngFromCoordinate)}
-            fillColor={MAP.POLYGON_COLOR}
-            color={MAP.POLYGON_COLOR}
-            interactive={false}
-        />
-    );
+    renderInactivePolygons = () =>
+        this.props.polygonCoordinates.map((coordinates, index) => {
+            return index === this.props.activePolygonIndex ? null : (
```

Comment:
We think that the intention of the function should be clear (we reformatted it to be even more clear). Since your suggestion iterates twice over the data for filtering only one element out of it, we would go with our solution since it performs better.